src/plugins/recordTypes/person/fields.js

```diff
@@ -1259,31 +1270,35 @@ export default (configContext) => {
                 },
               },
             },
-            suppliedBirthDate: {
+            suppliedStructuredBirthDateGroup: {
               [config]: {
+                dataType: DATA_TYPE_STRUCTURED_DATE,
                 messages: defineMessages({
                   fullName: {
-                    id: 'field.persons_common.suppliedBirthDate.fullName',
-                    defaultMessage: 'Birth date supplied',
+                    id: 'field.persons_common.suppliedStructuredBirthDateGroup.fullName',
+                    defaultMessage: 'Supplied structured birth date',
+                  },
+                  groupName: {
+                    id: 'field.persons_common.suppliedStructuredBirthDateGroup.groupName',
+                    defaultMessage: 'Date',
                   },
                   name: {
-                    id: 'field.persons_common.suppliedBirthDate.name',
+                    id: 'field.persons_common.suppliedStructuredBirthDateGroup.name',
                     defaultMessage: 'Supplied',
                   },
                 }),
-                repeating: true,
                 view: {
                   type: StructuredDateInput,
                 },
-                ...extensions.structuredDate.fields,
               },
+              ...extensions.structuredDate.fields,
```

Comment:
I wouldn't be surprised if this is the only reason the field didn't save, but the other changes are good to have too.